public/app/features/dataframe-import/utils.ts

```diff
@@ -0,0 +1,50 @@
+import { Accept } from 'react-dropzone';
+import { Observable } from 'rxjs';
+
+import { toDataFrame } from '@grafana/data';
+import { readSpreadsheet } from 'app/core/utils/sheet';
+
+import { FileImportResult } from './types';
+
+function getFileExtensions(acceptedFiles: Accept) {
+  const fileExtentions = new Set<string>();
+  Object.keys(acceptedFiles).forEach((v) => {
+    acceptedFiles[v].forEach((extension) => {
+      fileExtentions.add(extension);
+    });
+  });
+  return fileExtentions;
+}
+
+export function formatFileTypes(acceptedFiles: Accept) {
+  const fileExtentions = Array.from(getFileExtensions(acceptedFiles));
+  if (fileExtentions.length === 1) {
+    return fileExtentions[0];
+  }
+  return `${fileExtentions.slice(0, -1).join(', ')} or ${fileExtentions.slice(-1)}`;
+}
+
+export function filesToDataframes(files: File[]): Observable<FileImportResult> {
```

Comment:
Rather than supporting multiple async files, can we limit this to a single file at a time?  I don't think we want dropping a pile of csv files to generate a pile of table panels -- at least not to start!

Reply:
looks like the drop component enforces that, so maybe this is fine 🤷